libraries/AP_RTC/AP_RTC.cpp

```diff
@@ -125,12 +125,50 @@ bool AP_RTC::get_local_time(uint8_t &hour, uint8_t &min, uint8_t &sec, uint16_t
     uint32_t sec_ms = (ms_local % (60 * 1000)) - ms;
     uint32_t min_ms = (ms_local % (60 * 60 * 1000)) - sec_ms - ms;
     uint32_t hour_ms = (ms_local % (24 * 60 * 60 * 1000)) - min_ms - sec_ms - ms;
+    
 
     // convert times as milliseconds into appropriate units
     sec = sec_ms / 1000;
     min = min_ms / (60 * 1000);
     hour = hour_ms / (60 * 60 * 1000);
-
+    
+    // calculate human calendar date
+    const uint32_t secs_day   =  86400; //  24* 60 * 60
```

Comment:
we actually already have a function to parse dates here:
https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/hwdef/common/stm32_util.c#L144
I think we should move that function to AP_RTC, and then use the common function in both places. 
Is that something you are happy to do or should I do it?

Reply:
@tridge uh. i grew some grey hair trying to get that date-math right... is it ok if i try to move that existing function and cry for your help when i get stuck?